fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/postgresql/PostgreSqlCodeSystemDAO.java

```diff
@@ -21,84 +23,95 @@
+                if (log.isLoggable(Level.FINE)) {
+                        log.fine("DB read code system id complete. executionTime=" + dbCallDuration + "ms");
                 }
+                systemId = stmt.getInt(2);
             }
-            catch (SQLException e) {
-                if ("23505".equals(e.getSQLState())) {
-                    // another thread snuck in and created the record, so we need to fetch the correct id
-                    result = getCodeSystemId(codeSystem);
-
-                    if (result == null) {
-                        // This would be truly weird, but we protect against it anyway
-                        throw new IllegalStateException("No code system returned after duplicate found!");
-                    }
-                }
-                else {
-                    throw new FHIRPersistenceDataAccessException("codeSystem=" + codeSystem, e);
-                }
-            }
-
         }
-
-        // There's no way result can be null here, so we're OK returning an int
-        return result;
+        catch (Throwable e) {
+            throw new FHIRPersistenceDataAccessException(errMsg,e);
+        }
+        finally {
+            this.cleanup(null, getConnection());
+            log.exiting(CLASSNAME, METHODNAME);
+        }
+        return systemId;
     }
 
     /**
-     * Read the id for the named type
-     * @param codeSystem
-     * @return the database id, or null if the named record is not found
-     * @throws FHIRPersistenceDataAccessException
+     * Closes the passed PreparedStatement and Connection objects.
+     * @param stmt
+     * @param connection
      */
-    protected Integer getCodeSystemId(String codeSystem) throws FHIRPersistenceDataAccessException {
-        Integer result;
+    protected void cleanup(PreparedStatement stmt, Connection connection)  {
+        final String METHODNAME = "cleanup(PreparedStatement, Connection)";
+        log.entering(CLASSNAME, METHODNAME);
 
-        String sql1 = "SELECT code_system_id FROM code_systems WHERE code_system_name = ?";
+        FHIRPersistenceDBCleanupException ce;
 
-        try (PreparedStatement stmt = getConnection().prepareStatement(sql1)) {
-            stmt.setString(1, codeSystem);
-            ResultSet rs = stmt.executeQuery();
-            if (rs.next()) {
-                result = rs.getInt(1);
+        if (stmt != null) {
+            try {
+                stmt.close();
             }
-            else {
-                result = null;
+            catch (Throwable e) {
+                ce =  new FHIRPersistenceDBCleanupException("Failure closing PreparedStatement.",e);
+                log.log(Level.SEVERE, ce.getMessage(), ce);
             }
         }
-        catch (SQLException e) {
-            throw new FHIRPersistenceDataAccessException("codeSystem=" + codeSystem, e);
+        if(connection != null && this.getConnection() == null) {
+            try {
+                connection.close();
```

Comment:
Why are you forceabling closing this? It's part of a pool?

Reply:
this doesn't really close the connection, just return it to the pool. it's the same as what did for DB2.